src/applications/disability-benefits/all-claims/components/AutoComplete.jsx

```diff
@@ -0,0 +1,177 @@
+      const makePlural = resultCount > 1 ? 's' : '';
+
+      setAriaLiveText(
+        `${resultCount} result${makePlural}. ${freeTextResult}, (1 of ${resultCount})`,
+      );
+    }, 700),
+  ).current;
+
+  const debouncedSearch = useRef(
+    debounce(async inputValue => {
+      const freeTextResult = createFreeTextItem(inputValue);
+      const searchResults = fullStringSimilaritySearch(
+        inputValue,
+        availableResults,
+      );
+      const updatedResults = [freeTextResult, ...searchResults];
+      setResults(updatedResults);
+      setActiveIndex(0);
+
+      debouncedSetAriaLiveText(updatedResults.length, freeTextResult);
+    }, debounceDelay),
+  ).current;
+
+  const closeList = () => {
+    debouncedSearch.cancel();
+    debouncedSetAriaLiveText.cancel();
+    setResults([]);
+    setActiveIndex(null);
+  };
+
+  const handleInputChange = inputValue => {
+    setValue(inputValue);
+    onChange(inputValue);
+
+    if (!inputValue) {
+      closeList();
+      setAriaLiveText('Input is empty. Please enter a condition.');
+      return;
+    }
+
+    debouncedSearch(inputValue);
+  };
+
+  const scrollIntoView = index => {
+    const activeResult = resultsRef.current[index];
+    activeResult?.scrollIntoView({
+      block: 'nearest',
+    });
+  };
+
+  const navigateList = (e, adjustment) => {
+    e.preventDefault();
+    const newIndex = activeIndex + adjustment;
+
+    if (newIndex < 0 || newIndex > results.length - 1) {
+      return;
+    }
+
+    setActiveIndex(newIndex);
+    setAriaLiveText(
+      `${results[newIndex]}, (${newIndex + 1} of ${results.length})`,
+    );
+    scrollIntoView(newIndex);
+  };
+
+  const selectResult = result => {
+    const newValue = result === createFreeTextItem(value) ? value : result;
+    setValue(newValue);
+    onChange(newValue);
+    setAriaLiveText(`${newValue} is selected`);
+    closeList();
+
+    inputRef.current.shadowRoot.querySelector('input').focus();
+  };
+
+  const handleKeyDown = e => {
+    if (results.length > 0) {
+      if (e.key === 'ArrowDown') {
+        navigateList(e, 1);
+      } else if (e.key === 'ArrowUp') {
+        navigateList(e, -1);
+      } else if (e.key === 'Enter') {
+        selectResult(results[activeIndex]);
+      } else if (e.key === 'Escape') {
+        closeList();
+      }
+    }
+  };
+
+  return (
+    <div className="cc-autocomplete">
+      <VaTextInput
+        data-testid="autocomplete-input"
+        id={id}
+        label={label}
+        message-aria-describedby={!value ? INSTRUCTIONS : null}
+        ref={inputRef}
+        required
+        value={value}
+        onBlur={() => setTimeout(closeList, 200)} // Enables clicking option from list
+        onFocus={() => value && debouncedSearch(value)}
+        onInput={e => handleInputChange(e.target.value)}
+        onKeyDown={handleKeyDown}
+      />
+      {results.length > 0 && (
+        <ul
+          aria-hidden="true"
```

Comment:
I don't think that we'll want to aria hide the listbox all the time, just when it's not visible. Sighted screenreader users will be confused why they can't navigate to the ul and check out the contents before making a selection. Also having the aria-hidden here may mess with whether or not the component is shown on the forms shortcut pages

Reply:
Just noting that Will and I paired on this and came to the decision that this works. It feels so wrong to straight up put an aria-hidden true on an element we're interacting with , but the focus is technically staying in the input.